src/util/simplify_expr.cpp

```diff
@@ -153,6 +158,148 @@ simplify_exprt::simplify_popcount(const popcount_exprt &expr)
   return unchanged(expr);
 }
 
+/// Simplify String.endsWith function when argument are constant
+/// \param expr [in,out]: the expression to simplify
+/// \param ns: namespace
+/// \return: true iff expr was not modified
```

Comment:
The documentation should be updated to reflect changed interface using `resultt` (also for `isEmpty()` and `compareTo()`.

Reply:
done